internal/queryexecutor/range_query_planner.go

```diff
@@ -1,28 +1,43 @@
 package queryexecutor
 
 import (
-	"errors"
-
 	"github.com/IBM-Blockchain/bcdb-server/internal/stateindex"
 	"github.com/IBM-Blockchain/bcdb-server/pkg/constants"
 )
 
 type rangeQueryPlan struct {
 	startKey *stateindex.IndexEntry
 	endKey   *stateindex.IndexEntry
+	seekKey  *stateindex.IndexEntry
+}
+
+type toSeek interface {
+	Seek(key []byte) bool
 }
 
 func createQueryPlan(attribute string, conds *attributeTypeAndConditions) (*rangeQueryPlan, error) {
 	// we assume this function to get only valid conditions
 	//   - eq and no other conditions
 	//   - lt and lte do not appear together
 	//   - gt and gte do not appear together
+	//   - neq can appear alone or with lt, lte, gt, and gte
 
-	for c := range conds.conditions {
-		if c == constants.QueryOpNotEqual {
-			// TODO: support not equal
-			return nil, errors.New(constants.QueryOpNotEqual + " is not supported currently")
+	var seekKey *stateindex.IndexEntry
+	for c, v := range conds.conditions {
+		if c != constants.QueryOpNotEqual {
+			continue
 		}
+
+		seekKey = &stateindex.IndexEntry{
+			Attribute:     attribute,
+			Type:          conds.valueType,
+			ValuePosition: stateindex.Existing,
+			Value:         v,
+			KeyPosition:   stateindex.Ending,
+		}
```

Comment:
can we accumulate the $neq values to an array?

Reply:
Done